### ISSUE TITLE
`rustup_init.exe` unable to detect the install Microsoft C++ Build Tools

### ISSUE THREAD

**Issue by Marin-Kitagawa** [rust-lang/rust#112414]
# Bug explanation
I have MSVC++ $2022$ installed. Yet the setup is not detecting it. 

I expected to see this happen: *explanation*
I expect the installer to detect the installed MSVC tools

Instead, this happened: *explanation*
The setup wasn't able to detect it despite installing all of `Desktop Development with C++`. It clearly says $2013$ or later. I have $2022$ version $17.6.2$

# OS Specifications
```yaml
Edition	Windows: 11 Pro
Version:	22H2
OS build:	22621.1778
Experience:	Windows Feature Experience Pack 1000.22642.1000.0
```

I was trying to install Rust $1.70.0$
![image](https://github.com/rust-lang/rust/assets/49131888/fe1469f9-08b8-4d6f-8c06-db17a127b379)

**Comment by ChrisDenton** [comment]
Hm, I am not currently able to reproduce this on my Windows 11 machine (using a VM for a clean environment). I tried installing both the minimal necessary and also installing everything in Desktop C++. Either way rustup-init detected it and skipped that message.

<details><summary>A minimal selection installed</summary>

![vs_minimal](https://github.com/rust-lang/rust/assets/4459874/e9827ee4-dbab-461c-8f8c-0bc4cdd282b9)

</details>

Btw, this issue should be moved to the [rustup repository](https://github.com/rust-lang/rustup).
